src/fn/upload-custom-translations.js

```diff
@@ -81,6 +114,76 @@ function parse(filePath, options) {
   });
 }
 
+function checkTranslations(translations, lang, templatePlaceholders) {
+  let mf = null;
+  try {
+    mf = new MessageFormat(lang);
+  } catch (e) {
+    log.warn(`Cannot check '${lang}' translations: ${e.message}`);
+  }
+  const placeholders = extractPlaceholdersFromTranslations(translations);
+  let formatErrorsFound = 0,  placeholderErrorsFound = 0, emptiesFound = 0;
+  for (const [msgKey, msgSrc] of Object.entries(translations)) {
+    if (!msgSrc) {
+      emptiesFound++;
+    } else if (/{{[\s\w.#^/'|]+}}/.test(msgSrc)) {
+      if (templatePlaceholders) {
+        const placeholder = placeholders[msgKey];
+        if (placeholder) {
+          const templatePlaceholder = templatePlaceholders[msgKey];
+          if (!templatePlaceholder) {
+            log.error(`Cannot compile '${lang}' translation with key '${msgKey}' ` +
+              `has placeholders, but the key does not match any of ${EN_FILE}\n` +
+              'You can use messages-ex.properties to add placeholders missing from the reference context.');
+            placeholderErrorsFound++;
+          } else {
+            const foundAllPlaceholders = placeholder.every(el => templatePlaceholder.includes(el));
+            if (!foundAllPlaceholders) {
+              log.error(`Cannot compile '${lang}' translation with key '${msgKey}' ` +
+                `has placeholders that do not match those of ${EN_FILE}\n` +
+                'You can use messages-ex.properties to add placeholders missing from the reference context.');
+              placeholderErrorsFound++;
+            }
```

Comment:
I think we should allow this case. If the English file defines some placeholders but the project chooses not to use them that's ok. So I think you can remove this whole block.

Reply:
No, it is the opposite, if a non-english translation don't use one or all of the placeholders used in the `-en` translation is fine, this block detects when a non-english translation uses a placeholder that is not present in the `-en` version (or `-ex`).

Eg. this is OK:

**messages-en.properties**
```
Number\ in\ month = {{count}} in {{month}}
```

**messages-es.properties**
```
Number\ in\ month = {{count}} total
```

This is WRONG:

**messages-en.properties**
```
Number\ in\ month = {{count}} in {{month}}
```

**messages-es.properties**
```
Number\ in\ month = {{count}} en {{month}} de {{another_placeholder}}
```

`{{another_placeholder}}` needs to be defined in messages-en.properties or messages-ex.properties first.

Reply:
Ahh sorry! I read that wrong.